lib/turbo/turbo.dart

```diff
@@ -4,94 +4,207 @@ import 'package:ardrive/blocs/turbo_payment/utils/storage_estimator.dart';
 import 'package:ardrive/services/turbo/payment_service.dart';
 import 'package:ardrive/turbo/topup/models/price_estimate.dart';
 import 'package:ardrive/utils/data_size.dart';
+import 'package:ardrive/utils/disposable.dart';
 import 'package:ardrive/utils/file_size_units.dart';
 import 'package:ardrive/utils/logger/logger.dart';
 import 'package:arweave/arweave.dart';
 
-class Turbo {
-  Turbo({required this.paymentService, required this.wallet})
-      : _initialSessionTime = DateTime.now() {
-    _startSessionExpirationListener();
+class Turbo extends Disposable {
+  final TurboSessionManager _sessionManager;
+  final TurboCostCalculator _costCalculator;
+  final TurboBalanceRetriever _balanceRetriever;
+  final TurboPriceEstimator _priceEstimator;
+  final Wallet wallet;
+
+  Turbo({
+    required TurboCostCalculator costCalculator,
+    required TurboSessionManager sessionManager,
+    required TurboBalanceRetriever balanceRetriever,
+    required TurboPriceEstimator priceEstimator,
+    required this.wallet,
+  })  : _balanceRetriever = balanceRetriever,
+        _costCalculator = costCalculator,
+        _priceEstimator = priceEstimator,
+        _sessionManager = sessionManager;
+
+  DateTime get maxQuoteExpirationDate {
+    final maxQuoteExpirationTime = _priceEstimator.maxQuoteExpirationTime;
+
+    if (maxQuoteExpirationTime == null) {
+      throw Exception(
+          'maxQuoteExpirationTime is null. It can happen when you dont computate the balance first.');
```

Comment:
Is this something the user will see? 😬

Nit:
```suggestion
          "maxQuoteExpirationTime is null. It can happen when you don't compute the balance first.");
```